sandbox/src/main/java/com/badoo/ribs/sandbox/rib/switcher/SwitcherInteractor.kt

```diff
@@ -80,21 +81,36 @@ internal class SwitcherInteractor(
         backStack.pop()
     }
 
-    private val dialogEventConsumer : Consumer<Dialog.Event> = Consumer {
+    private val dialogEventConsumer: Consumer<Dialog.Event> = Consumer {
         when (it) {
             Dialog.Event.Positive -> {
                 // do something if you want
             }
         }
     }
 
-    private val transitionStateToViewModel : (Router.TransitionState) -> ViewModel = {
+    private val transitionStateToViewModel: (Router.TransitionState) -> ViewModel = {
         when (it) {
             SETTLED -> ViewModel(uiFrozen = false)
             IN_TRANSITION -> ViewModel(uiFrozen = true)
         }
     }
 
+    override fun onCreate(nodeLifecycle: Lifecycle) {
+        super.onCreate(nodeLifecycle)
+        whenChildBuilt<Menu>(nodeLifecycle) { commonLifecycle, child ->
+            commonLifecycle.createDestroy { bind(child.output to menuListener) }
+        }
```

Comment:
I could imagine a convenience API on top of this with some extension methods to make it a lot shorter:

```
withChild<Menu>.createDestroy { child ->
    bind(child.output to menuListener)
}
```

CommonLifecycle would be implicitly used. NodeLifecycle could be grabbed by plugin. I'm not yet sure about the case with viewLifecycle, but maybe another dedicated method could be added too?

Of course they would use these methods you added under the hood, so the more complex cases can be also supported if needed.

Reply:
Simplified API with default parameter for `nodeLifecycle`.

Reply:
That's a great first step - I wonder if we can also invoke `.createDestroy()` directly on `withChild<T>()`? Most of the noise comes from 1 extra lambda / case + 2 mentions of `commonLifecycle` / case.

Reply:
I realise that `.createDestroy` comes from MVICore, and we somehow still need the `child` param to go through that part, so maybe we can split it to two parts:

1. provde `withChild<T>()` to return some intermediary object that holds both `lifecycle` and `child`
2. provide `rib-mvicore` module that adds extension methods on this object

Reply:
I would prefer to create an extension method in our repository first to check how it works and feels. Simple inline method like:
```kotlin
inline fun <reified T1 : Rib, reified T2 : Rib> ChildAware.whenChildrenBuilt(
    lifecycle: Lifecycle = node.lifecycle,
    noinline callback: Binder.(T1, T2),
) {
    whenChildrenBuilt(lifecycle, T1::class, T2::class) { lifecycle.createDestroy { callback(c1, c2) } }
}
```